packages/contracts/contracts/CollateralManager.sol

```diff
@@ -0,0 +1,439 @@
+     * @param _collateralInfo Additional information about the collateral assets.
+     */
+    function checkBalances(
+        address _borrowerAddress,
+        Collateral[] calldata _collateralInfo
+    ) public returns (bool validated_, bool[] memory checks_) {
+        return _checkBalances(_borrowerAddress, _collateralInfo, false);
+    }
+
+    /**
+     * @notice Deploys a new collateral escrow and deposits collateral.
+     * @param _bidId The associated bidId of the collateral escrow.
+     */
+    function deployAndDeposit(uint256 _bidId) external onlyTellerV2 {
+        if (_isBidCollateralBacked[_bidId]) {
+            (address proxyAddress, ) = _deployEscrow(_bidId);
+            _escrows[_bidId] = proxyAddress;
+
+            for (
+                uint256 i;
+                i < _bidCollaterals[_bidId].collateralAddresses.length();
+                i++
+            ) {
+                _deposit(
+                    _bidId,
+                    _bidCollaterals[_bidId].collateralInfo[
+                        _bidCollaterals[_bidId].collateralAddresses.at(i)
+                    ]
+                );
+            }
+
+            emit CollateralEscrowDeployed(_bidId, proxyAddress);
+        }
+    }
+
+    /**
+     * @notice Gets the address of a deployed escrow.
+     * @notice _bidId The bidId to return the escrow for.
+     * @return The address of the escrow.
+     */
+    function getEscrow(uint256 _bidId) external view returns (address) {
+        return _escrows[_bidId];
+    }
+
+    /**
+     * @notice Gets the collateral info for a given bid id.
+     * @param _bidId The bidId to return the collateral info for.
+     * @return infos_ The stored collateral info.
+     */
+    function getCollateralInfo(uint256 _bidId)
+        public
+        view
+        returns (Collateral[] memory infos_)
+    {
+        CollateralInfo storage collateral = _bidCollaterals[_bidId];
+        address[] memory collateralAddresses = collateral
+            .collateralAddresses
+            .values();
+        infos_ = new Collateral[](
+            collateralAddresses.length
+        );
+        for (uint256 i; i < collateralAddresses.length; i++) {
+            infos_[i] = collateral.collateralInfo[collateralAddresses[i]];
+        }
+    }
+
+    /**
+     * @notice Withdraws deposited collateral from the created escrow of a bid.
+     * @param _bidId The id of the bid to withdraw collateral for.
+     */
+    function withdraw(uint256 _bidId) external {
+        if (_isBidCollateralBacked[_bidId]) {
+            BidState bidState = tellerV2.getBidState(_bidId);
+            require(
+                bidState == BidState.PAID ||
+                bidState == BidState.LIQUIDATED,
+                "Loan has not been repaid or liquidated"
+            );
+            address receiver;
+            if (bidState == BidState.LIQUIDATED) {
+                receiver = tellerV2.getLoanLender(_bidId);
```

Comment:
If the loan is liquidated, shouldn't the funds go to the liquidator wallet? The lender already had their loan repaid.